handler/src/main/java/io/netty/handler/pcap/PcapWriter.java

```diff
@@ -54,11 +61,18 @@ final class PcapWriter implements Closeable {
      *
      * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
      */
-    PcapWriter(OutputStream outputStream, ByteBuf byteBuf) throws IOException {
+    PcapWriter(OutputStream outputStream, ByteBuf byteBuf, boolean sharedOutputStream) throws IOException {
         this.outputStream = outputStream;
+        this.sharedOutputStream = sharedOutputStream;
 
         PcapHeaders.writeGlobalHeader(byteBuf);
-        byteBuf.readBytes(outputStream, byteBuf.readableBytes());
+        if (sharedOutputStream) {
+            synchronized(outputStream) {
```

Comment:
```suggestion
            synchronized (outputStream) {
```

We got to make checkstyle happy.